### ISSUE TITLE
poetry cannot run with PYTHONOPTIMIZE=2

### ISSUE THREAD

**Issue by allanlei** [sdispater/poetry#1039]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Docker python:3.6.8-alpine3.7
- **Poetry version**: 0.12.11, 0.12.12
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

## Issue

Setting `PYTHONOPTIMIZE=2` makes all `poetry` commands error with `The command name cannot be empty.`.  This includes a basic `poetry`.  Using `PYTHONOPTIMIZE=0 or 1` is OK.

`Dockerfile`
```
FROM        python:3.6.8-alpine3.7

ENV         PATH=/root/.poetry/bin:$PATH
ENV         POETRY_VERSION=0.12.12
ENV         PYTHONOPTIMIZE=0

RUN         apk add --no-cache --virtual .build-deps \
                curl

            # Install poetry
RUN         curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python
CMD         ["poetry", "-h"]
```

```
docker run -it --rm -e PYTHONOPTIMIZE=2 d54782d6c54f poetry -vvv

Traceback (most recent call last):
  File "/root/.poetry/bin/poetry", line 13, in <module>
    main()
  File "/root/.poetry/lib/poetry/console/__init__.py", line 5, in main
    return Application().run()
  File "/root/.poetry/lib/poetry/console/application.py", line 46, in __init__
    super(Application, self).__init__("Poetry", __version__)
  File "/root/.poetry/lib/poetry/_vendor/py3.6/cleo/application.py", line 66, in __init__
    for command in self.get_default_commands():
  File "/root/.poetry/lib/poetry/console/application.py", line 105, in get_default_commands
    commands = super(Application, self).get_default_commands()
  File "/root/.poetry/lib/poetry/_vendor/py3.6/cleo/application.py", line 547, in get_default_commands
    commands = [HelpCommand(), ListCommand()]
  File "/root/.poetry/lib/poetry/_vendor/py3.6/cleo/commands/command.py", line 58, in __init__
    super(Command, self).__init__(name or self.name)
  File "/root/.poetry/lib/poetry/_vendor/py3.6/cleo/commands/base_command.py", line 62, in __init__
    raise Exception('The command name cannot be empty.')
Exception: The command name cannot be empty.
```

**Comment by PhilipTrauner** [comment]
When `PYTHONOPTIMIZE=2` is enabled, docstrings are discarded, which isn't properly handled during command [signature resolution](https://github.com/sdispater/cleo/blob/cf44ac2eba2d6435516501e47e5521ee2da9115a/cleo/commands/command.py#L44). 
There doesn't appear to be an easy way around this, as `cleo` seems to be built around the assumption that `__doc__` attributes should be present and properly populated with a command description DSL (e.g. `clear { cache : The name of the cache to clear. } { --all : Clear all caches. }`)

**Comment by sdispater** [comment]
This is fixed in the `1.0.0` prereleases
